src/quo2/components/bottom_sheet/view.cljs

```diff
@@ -0,0 +1,174 @@
+            [react-native.platform :as platform]
+            [react-native.reanimated :as reanimated]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]))
+
+(def bottom-sheet-js (js/require "../src/js/bottom_sheet.js"))
+
+(def animation-delay 400)
+
+(defn with-animation [value & [options callback]]
+  (reanimated/with-spring
+    value
+    (clj->js (merge {:mass 2
+                     :stiffness 500
+                     :damping 200})
+             options)
+    callback))
+
+(defn get-bottom-sheet-gesture [pan-y translate-y bg-height bg-height-expanded window-height keyboard-shown disable-drag? expandable? show-bottom-sheet? expanded? close-bottom-sheet]
+  (-> (gesture/gesture-pan)
+      (gesture/on-start
+       (fn [_]
+         (when (and keyboard-shown (not disable-drag?) show-bottom-sheet?)
+           (re-frame/dispatch [:dismiss-keyboard]))))
+      (gesture/on-update
+       (fn [evt]
+         (when (and (not disable-drag?) show-bottom-sheet?)
+           (let [max-pan-up (if (or @expanded? (not expandable?))
+                              0
+                              (- (- bg-height-expanded bg-height)))
+                 max-pan-down (if @expanded?
+                                bg-height-expanded
+                                bg-height)]
+             (reanimated/set-shared-value pan-y (max
+                                                 (min
+                                                  (.-translationY evt)
+                                                  max-pan-down)
+                                                 max-pan-up))))))
+      (gesture/on-end
+       (fn [_]
+         (when (and (not disable-drag?) show-bottom-sheet?)
+           (let [end-pan-y (- window-height (.-value translate-y))
+                 expand-threshold (min (* bg-height * 1.1) (+ bg-height 50))
+                 collapse-threshold (max (* bg-height-expanded * 0.9) (- bg-height-expanded 50))
+                 should-close-bottom-sheet? (< end-pan-y (max (* bg-height 0.7) 50))]
+             (cond
+               should-close-bottom-sheet?
+               (close-bottom-sheet)
+
+               (and (not @expanded?) (> end-pan-y expand-threshold))
+               (reset! expanded? true)
+
+               (and @expanded? (< end-pan-y collapse-threshold))
+               (reset! expanded? false))))))))
+
+(defn bottom-sheet [props children]
+  (let [{on-cancel          :onCancel
+         disable-drag?      :disable-drag?
+         show-handle?       :show-handle?
+         visible?           :visible?
+         backdrop-dismiss?  :backdrop-dismiss?
+         expandable?        :expandable?
+         :or                {show-handle?       true
+                             backdrop-dismiss?  true
+                             expandable?        true}} props
+        content-height (reagent/atom nil)
+        visible-internal? (reagent/atom visible?)
```

Comment:
why `visible-internal?`

Reply:
simplified! thanks, for pointing that out :)